qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -71,6 +90,7 @@ def __init__(
         pulse_optimize: Union[bool, None] = None,
         natural_direction: Union[bool, None] = None,
         synth_gates: Union[List[str], None] = None,
+        method: Optional[str] = None,
```

Comment:
Bike shedding, but given strings are immutable, why not just `method: str = "default"`?  It'd also remove an `if/else` later down handling the lack of option.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/6124/commits/5d8539ac37300473270b470574aa8879c975e2f9